cmd/nerdctl/container_run.go

```diff
@@ -415,31 +408,27 @@ func createContainer(ctx context.Context, cmd *cobra.Command, client *containerd
 		args = newArg
 	}
 	var internalLabels internalLabels
-	internalLabels.platform = platform
+	internalLabels.platform = options.Platform
 
-	internalLabels.namespace = globalOptions.Namespace
+	internalLabels.namespace = options.GOptions.Namespace
 
 	var (
 		opts  []oci.SpecOpts
 		cOpts []containerd.NewContainerOpts
 		id    = idgen.GenerateID()
 	)
 
-	cidfile, err := cmd.Flags().GetString("cidfile")
-	if err != nil {
-		return nil, nil, err
-	}
-	if cidfile != "" {
-		if err := writeCIDFile(cidfile, id); err != nil {
+	if options.CidFile != "" {
+		if err := containerutil.WriteCIDFile(options.CidFile, id); err != nil {
 			return nil, nil, err
 		}
 	}
-	dataStore, err := clientutil.DataStore(globalOptions.DataRoot, globalOptions.Address)
+	dataStore, err := clientutil.DataStore(options.GOptions.DataRoot, options.GOptions.Address)
 	if err != nil {
 		return nil, nil, err
 	}
 
-	stateDir, err := getContainerStateDirPath(cmd, globalOptions, dataStore, id)
+	stateDir, err := containerutil.GetContainerStateDirPath(dataStore, id, options.GOptions.Namespace)
```

Comment:
nit: `GetContainerStateDirPath` -> `ContainerStateDirPath`. IIRC, the convention is to not have `Get` in getter funcs.